vividus-plugin-azure-storage-account/src/main/java/org/vividus/azure/storage/blob/BlobStorageSteps.java

```diff
@@ -231,6 +236,100 @@ public void findBlobs(StringComparisonRule rule, String blobNameToMatch, String
+     *                           <li><code>resultsLimit</code> Maximum number of blob names to return.</li>
+     *                          </ul>
+     * @param containerName     The name of the container to point to.
+     * @param storageAccountKey The key to Storage Account endpoint.
+     * @param scopes            The set (comma separated list of scopes e.g.: STORY, NEXT_BATCHES) of the variable
+     *                          scopes.<br>
+     *                          <i>Available scopes:</i>
+     *                          <ul>
+     *                          <li><b>STEP</b> - the variable will be available only within the step,
+     *                          <li><b>SCENARIO</b> - the variable will be available only within the scenario,
+     *                          <li><b>STORY</b> - the variable will be available within the whole story,
+     *                          <li><b>NEXT_BATCHES</b> - the variable will be available starting from next batch
+     *                          </ul>
+     * @param variableName      The variable name to store the list of found blob names.
+     */
+    @When("I filter blobs by:$filters in container `$containerName` of storage account `$storageAccountKey`"
+            + " and save result to $scopes variable `$variableName`")
+    public void findBlobsByFilters(List<BlobFilter> filters, String containerName,
+                                      String storageAccountKey, Set<VariableScope> scopes, String variableName)
+    {
+        List<BlobFilter> checkedFilters = checkFilters(filters);
+        List<String> blobNames = new ArrayList<>();
+        BlobContainerClient blobContainerClient = createBlobContainerClient(containerName, storageAccountKey);
+        for (BlobFilter filter : checkedFilters)
+        {
+            blobContainerClient
+                    .listBlobsByHierarchy(filter.getBlobNamePrefix().orElse(StringUtils.EMPTY))
+                    .stream()
+                    .map(BlobItem::getName)
+                    .filter(filterMatched(filter))
+                    .limit(filter.getResultsLimit().orElse(Long.MAX_VALUE))
+                    .forEach(blobNames::add);
+        }
+        bddVariableContext.putVariable(scopes, variableName, blobNames);
+    }
+
+    private static List<BlobFilter> checkFilters(List<BlobFilter> filters)
+    {
+        return filters.stream().map(filter -> {
+            if (filter.getBlobNameFilterRule() == null)
+            {
+                filter.setBlobNameFilterRule(Optional.empty());
+            }
+            if (filter.getBlobNameFilterValue() == null)
+            {
+                filter.setBlobNameFilterValue(Optional.empty());
+            }
+            if (filter.getBlobNamePrefix() == null)
+            {
+                filter.setBlobNamePrefix(Optional.empty());
+            }
+            if (filter.getResultsLimit() == null)
+            {
+                filter.setResultsLimit(Optional.empty());
+            }
```

Comment:
do not change the state of input parameters, it can be done via implementation of addition POJO API for convenience